### ISSUE TITLE
Memory overflow by Mahalanobis detector

### ISSUE THREAD

**Issue by kyoungup** [kkirchheim/pytorch-ood#9]
Hi,

The Mahalanobis detector is easy to experience memory shortage. In the example, CIFAR10 is used, which is 32x32. If we resize them to 224x224, which is de-facto in classificatio, it is easy to see memory overflow situation. Are there a way to calculate precision and etc. incrementally to avoid memory shortage?

**Comment by kkirchheim** [comment]
I think the easiest way to reduce memory would be to reduce the batch size of the data loader that you use to fit the model. 

Another method would be to extract the features manually and use the `fit_features()` method.

**Comment by kkirchheim** [comment]
Ah, I got it. This should be possible. In Scikit-Learn, there is iirc sometimes a method called `fit_partial()`. We could add something similar.

**Comment by kkirchheim** [comment]
I added a draft implementation in 5d8cdfaf9e58d24bb8ca5cce79efdc72ca18ebb5. However, it is not well tested yet. Would something like this work for you?

**Comment by kyoungup** [comment]
Right, fit_partial() is what I wanted. But, I will test it later. Thank you.

**Comment by kkirchheim** [comment]
Okay, it contains several bugs at the moment, I will see when I can fix it.
